--- a/src/base/process.cc	2021-05-29 22:40:50.000000000 +0900
+++ b/src/base/process.cc	2021-06-15 08:47:24.000000000 +0900
@@ -49,8 +49,33 @@
 #if defined(OS_LINUX) || defined(OS_ANDROID)
 #include <fcntl.h>
 #include <signal.h>
-#include <spawn.h>  // for posix_spawn().
+//#include <spawn.h>  // for posix_spawn().
 #include <sys/types.h>
+
+#include <unistd.h>
+#include <sys/wait.h>
+static int
+posix_spawn(pid_t *pid, const char *path,
+  const /*posix_spawn_file_actions_t*/void *fa,
+  const /*posix_spawnattr_t*/void *sa,
+  char * const argv[], char * const envp[])
+{
+  pid_t p;
+  volatile int error = 0;
+
+  p = vfork();
+  if (p == -1) return errno;
+  else if (p == 0) {
+    execve(path, argv, envp != NULL ? envp : environ);
+    error = errno;
+    _exit(127);
+  }
+  if (error != 0)
+    waitpid(p, NULL, WNOHANG);
+  else if (pid != NULL)
+    *pid = p;
+  return (error);
+}
 #endif  // OS_LINUX || OS_ANDROID
 
 #include <cstdlib>
