--- a/src/base/thread.cc
+++ b/src/base/thread.cc
@@ -34,6 +34,7 @@
 #include <windows.h>
 #else  // _WIN32
 #include <pthread.h>
+#include <signal.h>
 #endif  // _WIN32
 
 #include <atomic>
@@ -174,7 +175,7 @@ void Thread::Join() {
 
 namespace {
 
-#ifdef XXX__ANDROID__XXX
+#if 1
 
 void ExitThread(int sig) { pthread_exit(0); }
 
@@ -206,7 +207,7 @@ void InitPThreadCancel() {
   // Nothing is required.
 }
 
-void PThreadCancel(pthread_t thread_id) { pthread_cancel(thread_id); }
+void PThreadCancel(::pthread_t thread_id) { ::pthread_cancel(thread_id); }
 
 #endif  // XXX__ANDROID__XXX or others
 
