--- a/src/build_mozc.py	2021-05-29 22:40:50.000000000 +0900
+++ b/src/build_mozc.py	2021-06-15 08:32:49.000000000 +0900
@@ -264,9 +264,7 @@
     return dependencies + [meta_target_name]
 
   if target_platform == 'Linux':
-    targets = [SRC_DIR + '/server/server.gyp:mozc_server',
-               SRC_DIR + '/renderer/renderer.gyp:mozc_renderer',
-               SRC_DIR + '/gui/gui.gyp:mozc_tool']
+    targets = [SRC_DIR + '/server/server.gyp:mozc_server']
     if PkgExists('ibus-1.0 >= 1.4.1'):
       targets.append(SRC_DIR + '/unix/ibus/ibus.gyp:ibus_mozc')
   elif target_platform == 'Mac':
@@ -501,6 +499,7 @@
   # TODO(yukawa): Enable GYP_CROSSCOMPILE for Windows.
   if not IsWindows():
     os.environ['GYP_CROSSCOMPILE'] = '1'
+  gyp_options.extend(['-D', 'enable_gtk_renderer=0'])
 
   # Run GYP.
   logging.info('Running GYP...')
