--- a/src/build_mozc.py	2021-10-12 05:45:40.000000000 +0900
+++ b/src/build_mozc.py	2021-10-13 16:55:14.000000000 +0900
@@ -161,6 +161,11 @@
     if not PkgExists('ibus-1.0 >= 1.4.1'):
       logging.info('removing ibus.gyp.')
       gyp_file_names.remove('%s/unix/ibus/ibus.gyp' % SRC_DIR)
+    # Add fcitx.gyp/fcitx5.gyp if fcitx version is >=4.
+    if not PkgExists('fcitx >= 4'):
+      logging.info('removing fcitx.gyp.')
+      gyp_file_names.remove('%s/unix/fcitx/fcitx.gyp' % SRC_DIR)
+      gyp_file_names.remove('%s/unix/fcitx5/fcitx5.gyp' % SRC_DIR)
   gyp_file_names.sort()
   return gyp_file_names
 
@@ -264,9 +269,7 @@
     return dependencies + [meta_target_name]
 
   if target_platform == 'Linux':
-    targets = [SRC_DIR + '/server/server.gyp:mozc_server',
-               SRC_DIR + '/renderer/renderer.gyp:mozc_renderer',
-               SRC_DIR + '/gui/gui.gyp:mozc_tool']
+    targets = [SRC_DIR + '/server/server.gyp:mozc_server']
     if PkgExists('ibus-1.0 >= 1.4.1'):
       targets.append(SRC_DIR + '/unix/ibus/ibus.gyp:ibus_mozc')
   elif target_platform == 'Mac':
@@ -501,6 +504,7 @@
   # TODO(yukawa): Enable GYP_CROSSCOMPILE for Windows.
   if not IsWindows():
     os.environ['GYP_CROSSCOMPILE'] = '1'
+  gyp_options.extend(['-D', 'enable_gtk_renderer=0'])
 
   # Run GYP.
   logging.info('Running GYP...')
