--- a/src/build_mozc.py	2022-12-30 22:02:09
+++ b/src/build_mozc.py	2023-01-04 01:56:11
@@ -165,6 +165,11 @@
     if not PkgExists('ibus-1.0 >= 1.4.1'):
       logging.info('removing ibus.gyp.')
       gyp_file_names.remove('%s/unix/ibus/ibus.gyp' % SRC_DIR)
+    # Add fcitx.gyp/fcitx5.gyp if fcitx version is >=4.
+    if not PkgExists('fcitx >= 4'):
+      logging.info('removing fcitx.gyp.')
+      gyp_file_names.remove('%s/unix/fcitx/fcitx.gyp' % SRC_DIR)
+      gyp_file_names.remove('%s/unix/fcitx5/fcitx5.gyp' % SRC_DIR)
   gyp_file_names.sort()
   return gyp_file_names
 
@@ -272,8 +277,7 @@
 
   if target_platform == 'Linux':
     CheckGtkBuild(options)
-    targets = [SRC_DIR + '/server/server.gyp:mozc_server',
-               SRC_DIR + '/gui/gui.gyp:mozc_tool']
+    targets = [SRC_DIR + '/server/server.gyp:mozc_server']
     if USE_DEPRECATED_GTK_RENDERER:
       # Gtk candidate window built by mozc_renderer is no longer
       # included in the package alias.
@@ -568,6 +572,7 @@
   # TODO(yukawa): Enable GYP_CROSSCOMPILE for Windows.
   if not IsWindows():
     os.environ['GYP_CROSSCOMPILE'] = '1'
+  gyp_options.extend(['-D', 'enable_gtk_renderer=0'])
 
   # Run GYP.
   logging.info('Running GYP...')
