--- a/src/build_mozc.py.orig	2016-12-14 22:53:38.000000000 +0900
+++ b/src/build_mozc.py	2018-03-07 03:03:04.433146314 +0900
@@ -405,9 +405,7 @@
   if target_platform == 'Android':
     targets = [SRC_DIR + '/android/android.gyp:apk']
   elif target_platform == 'Linux':
-    targets = [SRC_DIR + '/server/server.gyp:mozc_server',
-               SRC_DIR + '/renderer/renderer.gyp:mozc_renderer',
-               SRC_DIR + '/gui/gui.gyp:mozc_tool']
+    targets = [SRC_DIR + '/server/server.gyp:mozc_server']
     if PkgExists('ibus-1.0 >= 1.4.1'):
       targets.append(SRC_DIR + '/unix/ibus/ibus.gyp:ibus_mozc')
   elif target_platform == 'Mac':
@@ -607,9 +603,6 @@
   else:
     gyp_options.extend(['-D', 'use_wix=NO'])
 
-  if target_platform == 'Linux':
-    gyp_options.extend(['-D', 'enable_gtk_renderer=1'])
-
   # Android
   if target_platform == 'Android':
     android_home = GetAndroidHome(options)
