--- a/mozc-config/mozc_config_main.cc
+++ b/mozc-config/mozc_config_main.cc
@@ -18,8 +18,8 @@
 
 using namespace std;
 
-#include "session/config.pb.h"
-#include "session/config_handler.h"
+#include "protocol/config.pb.h"
+#include "config/config_handler.h"
 #include "google/protobuf/descriptor.h"
 #include "google/protobuf/message.h"
 
@@ -94,7 +94,7 @@ void group_to_inner(string& s) {
     }
 }
 
-void get(string name) {
+void get_prop(string name) {
     mozc::config::Config conf;
     mozc::config::ConfigHandler::GetConfig(&conf);
 
@@ -148,7 +148,9 @@ void get(string name) {
 }
 
 void set_character_form_rules(string name, string group, string preedit, string conversion) {
-    mozc::config::Config conf = mozc::config::ConfigHandler::GetConfig();
+    mozc::config::Config conf;
+    mozc::config::ConfigHandler::GetConfig(&conf);
+    //mozc::config::Config conf = mozc::config::ConfigHandler::DefaultConfig();
 
     const google::protobuf::FieldDescriptor* field =
             mozc::config::Config::descriptor()->FindFieldByName(name);
@@ -201,7 +203,7 @@ void set_character_form_rules(string name, string group, string preedit, string
     mozc::config::ConfigHandler::SetConfig(conf);
 }
 
-void set(string name, string value) {
+void set_prop(string name, string value) {
     if (is_ignore(name)) return;
 
     mozc::config::Config conf;
@@ -246,10 +248,10 @@ void set(string name, string value) {
 void clear(string name) {
     if (is_ignore(name)) return;
 
-//    mozc::config::Config conf;
-  //  mozc::config::ConfigHandler::GetConfig(&conf);
+    mozc::config::Config conf;
+    mozc::config::ConfigHandler::GetConfig(&conf);
 
-    mozc::config::Config conf = mozc::config::ConfigHandler::GetConfig();
+    //mozc::config::Config conf = mozc::config::ConfigHandler::Config();
 
     const google::protobuf::FieldDescriptor* field =
             conf.GetDescriptor()->FindFieldByName(name);
@@ -288,10 +290,10 @@ void print_modifier() {
 }
 
 void print_all() {
-//    mozc::config::Config conf;
-//    mozc::config::ConfigHandler::GetConfig(&conf);
+    mozc::config::Config conf;
+    mozc::config::ConfigHandler::GetConfig(&conf);
 
-    mozc::config::Config conf = mozc::config::ConfigHandler::GetConfig();
+    //mozc::config::Config conf = mozc::config::ConfigHandler::GetConfig();
 
     const google::protobuf::Descriptor* desc = conf.GetDescriptor();//mozc::config::Config::descriptor();
     const google::protobuf::Reflection* ref = conf.GetReflection();
@@ -460,13 +462,13 @@ int main(int argc, char **argv) {
         print_modifier();
     } else if (flg == "-g") {
         if (argc > 2)
-            get((string)argv[2]);
+            get_prop((string)argv[2]);
     } else if (flg == "-s") {
         if ((string)argv[2] == "character_form_rules" && argc > 5) {
             set_character_form_rules((string)argv[2], (string)argv[3],
                                      (string)argv[4], (string)argv[5]);
         } else if (argc > 3) {
-            set((string)argv[2], (string)argv[3]);
+            set_prop((string)argv[2], (string)argv[3]);
         }
     } else if (flg == "-c") {
         if (argc > 2)
