--- /dev/null	2020-06-07 23:22:07.000000000 +0900
+++ b/src/google/protobuf/generated_message_table_driven.h	2020-06-07 22:18:34.000000000 +0900
@@ -0,0 +1,6 @@
+#ifdef __GNUC__
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-variable"
+#endif
+#define AuxillaryParseTableField AuxiliaryParseTableField  // in protobuf 3.12, the generated source has a spelling error
+#include_next <google/protobuf/generated_message_table_driven.h>
--- /dev/null	2020-06-08 01:53:10.000000000 +0900
+++ b/src/google/protobuf/port_def.inc	2020-06-08 01:52:27.000000000 +0900
@@ -0,0 +1,4 @@
+#include_next <google/protobuf/port_def.inc>
+
+#undef PROTOBUF_FINAL
+#define PROTOBUF_FINAL
